### PR TITLE
experimental codeql workflow

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

...

### Solution

...

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, master-2.x ]
```
Comment (AnnaShaleva) [inline .github/workflows/codeql-analysis.yml:16]: Do we really need it on push?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
```
Comment (AnnaShaleva) [inline .github/workflows/codeql-analysis.yml:18]: Just a question, why can't we just add
```
on:
    pull_request:
        branches: [ master, master-2.x ]
```

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
```
Comment (AnnaShaleva) [inline .github/workflows/codeql-analysis.yml:31]: Then don't we want to specify the go version?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (roman-khimov) [reply .github/workflows/codeql-analysis.yml:16]: I think it needs some reference point to compare new PRs with. Frankly, this file is autogenerated by GitHub, I just wanted to see whether it finds anything and it actually did find some (very minor, but still) things to improve.

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (roman-khimov) [reply .github/workflows/codeql-analysis.yml:31]: I don't think any version is to be specified there, Go itself is stable.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**